### ISSUE TITLE
`assert_never()` should not be flagged unreachable or I should be able to ignore it

### ISSUE THREAD

**Issue by cb0s** [DetachHead/basedpyright#1676]
### Description

When I am writing something like [this](https://basedpyright.com/?typeCheckingMode=standard&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFOQM71kwD6KJAbmQFDcAmJYFGBgwACnIAuQsTJUA2gCJyigLoBKKAFoAfFHowQk7lFOYh5KAF4rUZYuNmnUECRgBXECjtwkCBHCKJmYklEyOzqYMTHhsnGRiigDKkG4AFqgYAO4kKPhZ4OgAdCWKmgDEUAEgSGhpMNK1KLgk8q44eACqKK7kAMZp5ABGlCSq3EA) I don't want it to warn me about unreachable code. Even further: if I specifically tell it to ignore the case it should stop warning me about it.

The general issue with `assert_never()` also exists upstream (cf. [#11120](https://github.com/microsoft/pyright/issues/11120) and [this playground code](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFOQM71kwD6KJAbmQFDcAmJYFGBgwACnIAuQsTJUA2gCJyigLoBKKAFoAfFHowQk7lFOYh5KAF4rUZYuNmnUECRgBXECjtwS5ABaKJmYklEyOzqYMTHhsnGRiigDKkG7%2BqBgA7iQo%2BJng6AB0xYqaAMRQCHAgSGj%2BMNK1KLgk8q44eACqKK7kAMb%2B5ABGlCSq3EA)) but they decided to not consider it a bug. However I am a bit surprised why the `# pyright: ignore[reportUnreachable]` has no effect in this case.
If you want me to open this upstream as well, please tell me.

I am using neovim 0.11.4 with basedpyright 1.36.0, but as you can see, I also implemented it in your test environments and it also exists there.

Again the code to reproduce in case the sandbox environment links don't work:

```Python
from typing import Literal, assert_never

def foo(a: Literal["a"]) -> str:
    if a == "a":
        return "yeah"
    else:
        assert_never("Something went wrong...") # pyright: ignore[reportUnreachable]
```

**Comment by DetachHead** [comment]
basedpyright already special cases `assert_never` with `reportUnreachable`. the issue here is that you're using `assert_never` incorrectly. it's supposed to take the narrowed value as its argument, not an error message.

Code sample in [basedpyright playground](https://basedpyright.com/?pyrightVersion=1.35.0&typeCheckingMode=standard&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFOQM71kwD6KJAbmQFDcAmJYFGBgwACnIAuQsTJUA2gCJyigLoBKKAFoAfFHowQk7lFOYh5KAF4rUZYuNmnUECRgBXECjtwkCBHCKJmYklEyOzqYMTHhsnGQS6txAA)

```python
from typing import Literal, assert_never

def foo(a: Literal["a"]) -> str:
    if a == "a":
        return "yippy"
    else:
        assert_never(a)  # no error
```

**Comment by cb0s** [comment]
You are right, I am sorry for wasting your time. Never crossed my mind that I could have misunderstood this part...

**Comment by KotlinIsland** [comment]
`assert_never` is stupid, why doesn't it allow you to provide a message????
